nens_gs_uploader.py

```diff
@@ -0,0 +1,374 @@
+            "username": setting.username,
+            "password": setting.password,
+        }
+        # Layer
+        setting.in_layer = in_path
+
+        # Postgis single sld.
+        if setting.use_single_sld:
+            setting.in_sld_path = setting.postgis_sld_path
+
+        elif setting.use_batch_sld:
+            setting.in_sld_path = setting.postgis_sld_generator()
+
+        elif not setting.use_standard_sld:
+            print("Choose 'use_single_sld', 'use_batch', 'use_standard_sld'")
+
+        else:
+            pass
+
+    elif setting.use_shape:
+        # Datasource
+        setting.in_datasource = in_path
+
+        # Layer
+        setting.in_layer = None
+
+        # Styling
+        setting.in_sld_path = in_path.replace(".shp", ".sld")
+
+    else:
+        print("use either use_postgis or use_batch")
+
+    # General stuff
+    metadata = {
+        "pg_layer": layer_name,
+        "gs_workspace": workspace_name,
+        "gs_store": store_name,
+        "gs_layer": layer_name,
+        "uploader": setting.eigen_naam.lower(),
+        "projectnummer": setting.project_nummer.lower(),
+        "einddatum": setting.einddatum,
+    }
+
+    # Overwrite sld path
+    if setting.use_standard_sld:
+        styles = ["begaanbaarheid_wegen", "kwetsbaarheid_panden", "kwetsbare_objecten"]
+        list_style = [style for style in styles if style in onderwerp]
+        setting.in_sld_path = get_standard_sld(list_style[0])
+
+    # Overwrite all
+    if setting.overwrite_all:
+        setting.overwrite_postgres = True
+        setting.overwrite_abstract = True
+        setting.overwrite_feature = True
+        setting.overwrite_sld = True
+
+    # Outputs
+    setting.workspace_name = workspace_name
+    setting.store_name = store_name
+    setting.layer_name = layer_name
+    setting.abstract_data = abstract_data
+    setting.metadata = metadata
+    setting.subject = onderwerp
+
+    return setting
+
+
+def batch_upload(inifile):
+    """ Returns batch upload shapes for one geoserver """
+    setting = settings_object(inifile)
+
+    # set logging
+    set_log_config(setting.ini_location)
+    sys.stdout = logger(setting.ini_location)
+
+    # get vectors
+    if setting.use_shape:
+        in_paths = glob(setting.directory + "/*.shp")
+        subjects = [get_subject_from_path(in_path) for in_path in in_paths]
+
+    elif setting.use_postgis:
+        in_paths, subjects = setting.get_postgis_batchlayers()
+
+    else:
+        print("use either use_postgis or use_batch")
+
+    print_list(in_paths, "Paths")
+    print_dictionary(setting.__dict__, "Settings")
+
+    failures = {}
+    succes = {}
+    for count, (in_path, subject) in enumerate(zip(in_paths, subjects)):
+        log_time("info", percentage(count, len(in_paths)), subject, "l")
+
+        # set last settings
+        setting.server = wrap_geoserver(setting.server_naam)
+        setting = add_output_settings(setting, subject, in_path)
```

Comment:
- setting enkelvoud/meervoud.
- Ik zou verwacht hebben dat je settings aan het settingsobject toe zou voegen qua naam. Maar hier wordt setting overschreven. Even kijken of dit helemaal goed gaat.